services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbServiceUtils.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.deviceregistry.mongodb.utils;
+
+import java.util.Optional;
+
+import org.eclipse.hono.deviceregistry.mongodb.model.BaseDto;
+
+import com.mongodb.ErrorCategory;
+import com.mongodb.MongoException;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.mongo.MongoClient;
+
+@SuppressWarnings( {"checkstyle:MissingJavadocType", "checkstyle:HideUtilityClassConstructor"})
+/**
+ * Utility class for common functions across device registry services.
+ */
+public final class MongoDbServiceUtils {
+
+    /**
+     * Checks if a provided error is {@link MongoException} with error category {@link ErrorCategory#DUPLICATE_KEY}.
+     *
+     * @param throwable The exception to check. Other Throwables but {@link MongoException} return {@code false}.
+     * @return {@code true} if exception is an {@link ErrorCategory#DUPLICATE_KEY} exception.
+     */
+    public static boolean ifDuplicateKeyError(final Throwable throwable) {
+        if (throwable instanceof MongoException) {
+            final MongoException mongoException = (MongoException) throwable;
+            return ErrorCategory.fromErrorCode(mongoException.getCode()) == ErrorCategory.DUPLICATE_KEY;
+        }
+        return false;
+    }
+
+    /**
+     * Check if a Version is different or not set.
+     *
+     * @param expectedVersion new version, can unset
+     * @param actualValue     current version
+     * @return {@code true}, if different version or {@code expectedVersion} is unset
+     */
+    public static boolean isVersionDifferent(@SuppressWarnings("OptionalUsedAsFieldOrParameterType") final Optional<String> expectedVersion, final String actualValue) {
```

Comment:
Are you working on the above too?

Reply:
> This is being used in only one place inside `isVersionEqualCurrentVersionOrNotSet(..)`. FMPOV there is no need of a separate public static method for that. Also there is already a method `isVersionMatch(...)` doing the same in `Versioned` class. IMHO this can be removed.
You're right. done.